src/Development/IDE/GHC/Compat.hs

```diff
@@ -325,6 +326,31 @@ dontWriteHieFiles d =
     d
 #endif
 
+setUpTypedHoles ::DynFlags -> DynFlags
+#if MIN_GHC_API_VERSION(8,6,0)
+setUpTypedHoles df
+  = flip gopt_unset Opt_AbstractRefHoleFits    -- too spammy
+#if MIN_GHC_API_VERSION(8,8,0)
+  $ flip gopt_unset Opt_ShowDocsOfHoleFits     -- not used
+#endif
+  $ flip gopt_unset Opt_ShowMatchesOfHoleFits  -- nice but broken (forgets module qualifiers)
+  $ flip gopt_unset Opt_ShowProvOfHoleFits     -- not used
+  $ flip gopt_unset Opt_ShowTypeAppOfHoleFits  -- not used
+  $ flip gopt_unset Opt_ShowTypeAppVarsOfHoleFits -- not used
+  $ flip gopt_unset Opt_ShowTypeOfHoleFits     -- massively simplifies parsing
+  $ flip gopt_set   Opt_SortBySubsumHoleFits   -- very nice and fast enough in most cases
```

Comment:
This line slows the whole thing by a lot, since it does pairwise comparison between every single possible fit, and everything grinds to a halt. I'd *strongly* recommend you turn it off!

Reply:
You mean sort by subsumption right? It wasn't too bad in my empirical tests. It's quadratic complexity but the number of items is small, so only in degenerate cases like `_ :: a` does it grind to a halt, and ghcide will cancel it as soon as an edit is made anyway.

Overall, there is no set of flags that will make everyone happy, so we should eventually expose this flags as a user configurable setting.